File: app/src/main/java/com/example/tutorial/MainActivity.java
package com.example.tutorial;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.graphics.Color;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.LinearLayout;
import android.widget.RelativeLayout;
import android.widget.TextView;

import com.google.android.material.floatingactionbutton.FloatingActionButton;

import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {
    private TextView textView;
    private LinearLayout currentLayout;
    private Button button1, button2, button3, button4, button5, button6, button7, button8;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        //TextViews
        textView = findViewById(R.id.hello_world);

        //Buttons
        button1 = findViewById(R.id.button);
        button2 = findViewById(R.id.button2);
        button3 = findViewById(R.id.button3);
        button4 = findViewById(R.id.button4);
        button5 = findViewById(R.id.button5);
        button6 = findViewById(R.id.button6);
        button7 = findViewById(R.id.button7);
        button8 = findViewById(R.id.button8);

        //Layouts
        currentLayout = findViewById(R.id.main_layout);

        button1.setOnClickListener(v -> {
            Intent intent = new Intent(".FileActivity");
            startActivity(intent);
        });
        button2.setOnClickListener(v -> {
            Intent intent = new Intent(".SensorActivity");
            startActivity(intent);
        });
        button3.setOnClickListener(v -> {
            Intent intent = new Intent(".FragmentActivity");
            startActivity(intent);
        });
        button4.setOnClickListener(v -> {
            Intent intent = new Intent(".DrawingActivity");
            startActivity(intent);
        });
        button5.setOnClickListener(v -> {
            textView.setText(button5.getText());
            currentLayout.setBackgroundResource(R.color.purple_200);

        });
        button6.setOnClickListener(v -> {
            textView.setText(button6.getText());
            currentLayout.setBackgroundResource(R.color.red);

        });
        button7.setOnClickListener(v -> {
            textView.setText(button7.getText());
            currentLayout.setBackgroundResource(R.color.orange);

        });
        button8.setOnClickListener(v -> {
            textView.setText(button8.getText());
            currentLayout.setBackgroundResource(R.color.yellow);

        });

        //end
    }


    public void reset(View view) {
        currentLayout = (LinearLayout) findViewById(R.id.main_layout);
        textView.setText(R.string.hello);
        Log.i("Hello_world", "Hello pressed");
        currentLayout.setBackgroundResource(R.color.white);
    }

    @Override
    protected void onStart() {
        super.onStart();
        Log.i("LifeActivity###", "onStart");
    }

    @Override
    protected void onResume() {
        super.onResume();
        Log.i("LifeActivity###", "onResume");

    }

    @Override
    protected void onStop() {
        super.onStop();
        Log.i("LifeActivity###", "onStop");
        textView.setText("World!!!!!!!!");

    }

    @Override
    protected void onRestart() {
        super.onRestart();
        Log.i("LifeActivity###", "onRestart");

    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        Log.i("LifeActivity###", "OnCreate");
    }

}

File: app/src/main/java/com/cruelgeroi/apphub/activities/MainActivity.kt
package com.cruelgeroi.apphub.activities

import android.app.AlertDialog
import android.app.Dialog
import android.content.DialogInterface
import android.content.Intent
import android.os.Build
import android.os.Bundle
import android.util.Log
import android.view.Menu
import android.view.MenuItem
import android.view.View
import android.widget.Toast
import androidx.annotation.RequiresApi
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.DialogFragment
import com.cruelgeroi.apphub.R
import com.google.android.material.textfield.TextInputEditText


class MainActivity : AppCompatActivity() {


    private lateinit var mIntent:Intent
    private lateinit var mEditText: TextInputEditText

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        //Initializing
        val toolbar:androidx.appcompat.widget.Toolbar? = findViewById(R.id.toolbar)
        setSupportActionBar(toolbar)
    }

    override fun onCreateOptionsMenu(menu: Menu?): Boolean {
        menuInflater.inflate(R.menu.menu_activity_main, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        return when(item?.itemId){
            R.id.secret_button -> {
                val dialogFragment = CustomDialogFragment()
                dialogFragment.show(supportFragmentManager, "custom")
                true
            }
            else -> {
                super.onOptionsItemSelected(item)
            }
        }

    }

    fun onClickBtn(v: View?) {
        when(v?.id){
            R.id.button -> {
                startSelectedActivity(".FileActivity")
            }
            R.id.button2 -> {
                startSelectedActivity(".SensorActivity")
            }
            R.id.button3 -> {
                startSelectedActivity(".FragmentActivity")
            }
            R.id.button4 -> {
                startSelectedActivity(".DrawingActivity")
            }
            R.id.button5 -> {
                startSelectedActivity(".CircleActivity")
            }
            R.id.button6 -> {
                startSelectedActivity(".CalculatorActivity")
            }
            R.id.button7 -> {
                startSelectedActivity(".RecyclerActivity")
            }
            R.id.button8 -> {
                startSelectedActivity(".DatabaseActivity")
            }
        }
    }

    private fun startSelectedActivity(selectedIntent: String){
        mIntent = Intent(selectedIntent)
        startActivity(mIntent)
    }

    class CustomDialogFragment : DialogFragment() {
        @RequiresApi(Build.VERSION_CODES.LOLLIPOP)
        override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
            val requiredPassword:String? = getString(R.string.secret_password)
            val mIntent = Intent(".SecretActivity")

            val builder = AlertDialog.Builder(activity, R.style.AlertDialogTheme)
            val inflater = requireActivity().layoutInflater

            val dialogView = inflater.inflate(R.layout.dialog_secret, null)
            val passwordEditText = dialogView.findViewById<TextInputEditText>(R.id.secret_password_et)

            return builder
                .setTitle(R.string.secret_dialog_title)
                .setView(dialogView)
                .setPositiveButton(R.string.dialog_positive_button_text,
                    DialogInterface.OnClickListener { dialog, id ->
                        var password:String? = passwordEditText.text.toString()
                        if (requiredPassword == password) startActivity(mIntent)
                        else Toast
                            .makeText(inflater.context, "Its not working. Required password: '$requiredPassword'. Given password: '$password' ", Toast.LENGTH_SHORT)
                            .show()
                    })
                .setNegativeButton(R.string.dialog_negative_button_text,
                    DialogInterface.OnClickListener { dialog, id ->
                        dialog.dismiss()
                    })
                .create()
        }
    }
}